### PR TITLE
[MM-9703] Push notifications are sent for own DM's

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
Push notifications are sent for own DM's.
If you receive a webhook notification it will send push/email notification. if you write your own messages it will not send any notification.

#### Ticket Link
Jira: https://mattermost.atlassian.net/browse/MM-9703

### CODE REVIEW

**Comment by koxen** [comment]
:( I really enjoyed this behaviour while testing some push notification changes... ;D

File: app/notification.go [hunk 1]
```diff
@@ -55,10 +55,15 @@ func (a *App) SendNotifications(post *model.Post, team *model.Team, channel *mod
...
 	if channel.Type == model.CHANNEL_DIRECT {
 		var otherUserId string
-		if userIds := strings.Split(channel.Name, "__"); userIds[0] == post.UserId {
-			otherUserId = userIds[1]
-		} else {
-			otherUserId = userIds[0]
+
+		userIds := strings.Split(channel.Name, "__")
+
+		if strings.Compare(userIds[0], userIds[1]) != 0 {
```
Comment (koxen) [inline app/notification.go]: why not userIds[1] != userIds[0] ?

Comment (cpanato) [reply app/notification.go]: also works :) thanks for the review :)

**Comment by lindy65** [comment]
Thanks @cpanato @jasonblais 

I've tested the PR on the above spinmint and it's working as expected.  No notification is received when sending own DM messages.

**Comment by cpanato** [comment]
@enahum rebased

**Pull request merged**